scipy/special/cephes/hyp2f1.c

```diff
@@ -590,3 +601,30 @@ static double hyp2f1ra(double a, double b, double c, double x,
 
     return f0;
 }
+
+
+/*
+    15.4.2 Abramowitz & Stegun.
+*/
+static double hyp2f1_neg_c_equal_bc(double a, double b, double x)
+{
+    /*
+    	"converting floating point to integers is shockingly expensive"
+	
+	Therefore, we stay with double the entire time.
+    */
+    double k;
+    double sum = 0;
+    double x_k = 1; // x^k
+    double k_f = 1; // k!    
+
+    for (k = 0; k <= -b; k++) {
+	if (k != 0) {
+	    k_f *= k;  // We have a single CPU pipeline loss (not bad!)
+	}
+	sum += poch(a, k) * x_k / k_f;
```

Comment:
The factorial will overflow quickly so you don't want to explicitly compute it. Instead have a variable like `term` which contains the current term and then gets updated at every step by multiplying by `(a + k)*x/k` (probably some index errors there but you get the idea).

Reply:
As I said, `k < 100` in the worst-case scenario.

Reply:
Wait, that's me being idiotic.